dune/grid/io/file/printgrid.hh

```diff
@@ -0,0 +1,191 @@
+    plotfile << p2[0] << "," << p2[1] << " to ";
+    plotfile << p1[0] << "," << p1[1];
+    plotfile << " " << options << std::endl;
+  }
+
+}
+
+/** \brief Print a grid as a gnuplot for testing and development
+ *  \tparam GridType the type of grid to work with
+ *  \param grid the grid to print
+ *  \param helper an MPIHelper to create unique output file names in parallel case
+ *  \param output_file the base of the output filename
+ *  \param size size of the plot in pixels; increase if plot is too cramped
+ *  \param execute_plot whether to execute gnuplot automatically
+ *  \param png whether to use PNG or SVG as the output format
+ *  \param local_corner_indices whether to show local corner indices
+ *  \param local_intersection_indices whether to show local intersection indices
+ *  \param outer_normals whether to show outer normals of intersections
+ *  Creates a gnuplot (one per process if parallel) showing the grid structure with indices, intersection types etc.
+ */
+template <typename GridType>
+void printgrid (const GridType& grid, const Dune::MPIHelper& helper, std::string output_file = "printgrid",
+                int size = 2000, bool execute_plot = true, bool png = true, bool local_corner_indices = true,
+                bool local_intersection_indices = true, bool outer_normals = true)
+{
+
+  // Create output file
+  output_file = output_file + "_" + std::to_string(helper.rank());
+  std::ofstream plotfile (output_file, std::ios::out | std::ios::trunc);
+  if (!plotfile.is_open()) {
+    std::cerr << "Could not create plot file" << std::endl;
+    return;
+  }
+
+  // Basic plot settings
+  plotfile << "set size ratio -1" << std::endl;
+  if (png) {
+    plotfile << "set terminal png size " << size << "," << size << std::endl;
+    plotfile << "set output '" << output_file << ".png'" << std::endl;
+  } else {
+    plotfile << "set terminal svg size " << size << "," << size << " enhanced background rgb 'white'" << std::endl;
+    plotfile << "set output '" << output_file << ".svg'" << std::endl;
+  }
+
+  // Get GridView
+  typedef typename GridType::LeafGridView GV;
+  const GV& gv=grid.leafGridView();
+
+  // Create mappers used to retrieve indices
+  typedef typename Dune::LeafMultipleCodimMultipleGeomTypeMapper<GridType, ElementDataLayout      > ElementMapper;
+  typedef typename Dune::LeafMultipleCodimMultipleGeomTypeMapper<GridType, NodeDataLayout         > NodeMapper;
+  ElementMapper elementmapper(grid);
+  NodeMapper nodemapper(grid);
+
+  // Create iterators
+  typedef typename GV::template Codim<0 >::Iterator LeafIterator;
+  typedef typename GV::IntersectionIterator IntersectionIterator;
+
+  LeafIterator it = gv.template begin<0>();
+
+  // Will contain min/max coordinates. Needed for scaling of the plot
+  Dune::FieldVector<typename GridType::ctype,2> max_coord (it->geometry().center()), min_coord (max_coord);
+
+  // Iterate over elements
+  for (; it != gv.template end<0>(); ++it) {
+
```

Comment:
Due to the way entities work now, it would be better to get the entity once at the beginning of the loop and then use that:
```c++
const auto& e = *it;
...
e.geometry();
```